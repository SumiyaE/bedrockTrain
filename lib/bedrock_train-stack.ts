import * as cdk from 'aws-cdk-lib';
import {Construct} from 'constructs';
import {NodejsFunction} from "aws-cdk-lib/aws-lambda-nodejs";
import {Architecture, Runtime, Tracing} from "aws-cdk-lib/aws-lambda";
import {Effect, PolicyStatement} from "aws-cdk-lib/aws-iam";

export class BedrockTrainStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);
    const bedRockAccessPolicy = new PolicyStatement({
      effect: Effect.ALLOW,
      actions: ['bedrock:InvokeModel'],
      resources: ['*'],
    })

    new NodejsFunction(this, 'BedrockTrainLambda', {
      entry: 'lambda/entryPoint/logSurvey.ts',
      handler: 'handler',
      runtime: Runtime.NODEJS_LATEST,
      architecture: Architecture.ARM_64,
      initialPolicy: [bedRockAccessPolicy],
      timeout: cdk.Duration.seconds(30),
      tracing: Tracing.ACTIVE,
    })
  }
}
