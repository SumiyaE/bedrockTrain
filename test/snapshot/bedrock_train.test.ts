import {expect, test} from "vitest";
import * as cdk from "aws-cdk-lib";
import {Template} from "aws-cdk-lib/assertions";
import {ignoreAssetHashSerializer} from "../plugins/ignore-asset-hash";
import {BedrockTrainStack} from "../../lib/bedrock_train-stack";

test('snapshot test', () => {
  const app = new cdk.App();
  const stack = new BedrockTrainStack(app, 'BedrockTrainStackTest', {});
  const template = Template.fromStack(stack);

  // 次の行でプラグインをsnapshotSerializerに登録する
  expect.addSnapshotSerializer(ignoreAssetHashSerializer); // ★追加
  expect(template).toMatchSnapshot();
});
