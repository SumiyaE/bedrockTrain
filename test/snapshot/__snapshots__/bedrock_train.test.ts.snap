// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "BedrockTrainLambda3FDE6CA3": {
      "DependsOn": [
        "BedrockTrainLambdaServiceRoleDefaultPolicyA05D87BE",
        "BedrockTrainLambdaServiceRole82418156",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "HASH-REPLACED.zip",
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "BedrockTrainLambdaServiceRole82418156",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 30,
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "BedrockTrainLambdaServiceRole82418156": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "BedrockTrainLambdaServiceRoleDefaultPolicyA05D87BE": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "bedrock:InvokeModel",
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "BedrockTrainLambdaServiceRoleDefaultPolicyA05D87BE",
        "Roles": [
          {
            "Ref": "BedrockTrainLambdaServiceRole82418156",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
